### DISCUSSION TITLE
Best values for the ReBar Settings in NV Profile Inspector?

### DISCUSSION THREAD

**Discussion by Wardog19** [terminatorul/NvStrapsReBar#70] [Q&A] [answered]
Hey!

I am curious about the best settings in the Nvidia Profile Inspector, to assure the best performance for the ReBar Mod.
My Setup is a RTX 2080 Super with 8 gb VRAM and 8 gb ReBar enabled.

I have these settings right now:
![ReBar_NVPI](https://github.com/terminatorul/NvStrapsReBar/assets/151050552/903ab037-f0e8-47fe-a4d1-4411760d599e)

Do u guys think it would be better to put the Limit = VRAM, or even above?
Normally, this setting is completely blank.

And do i even need to set the rBAR - Options?

What is your experience with the settings?

These are the Size Limits available: 

0x000000000C800000 = 200mb

0x0000000012C00000 = 300mb 

0x0000000040000000 = 1gb (default value for most whitelisted games)

0x0000000060000000 = 1.5gb

0x0000000080000000 = 2gb

0x0000000100000000 = 4gb

0x0000000200000000 = 8gb

0x0000000280000000 = 10gb

**Comment by LsdMax** [comment]
It is unnecessary to make more than 1 gigabyte for games.
The reason is that only 4 requests of 256 megabytes are remembered.
If you do more, you will have overflowing memory in the video card. 
For video editing programs, you can make more than 1 gigabyte.

**Reply by Wardog19** [reply]
Thank you! 
So i have to leave everything as it is, except for the limit option?

**Reply by LsdMax** [reply]
I have these settings, but I don’t know if this is correct or not.
1. ReBar feature - ENABLED
2. ReBar options - 0x00000001.
3. ReBar size limit - 0x0000000040000000.

**Comment by terminatorul** [comment, marked as answer]
The best value depends on the game title.

I would go for max BAR size for your GPU in the global profile (default profile). I don't know what the rBAR options do, so I would not enable that blindly for the global profile, just enable the right BAR size instead.

For specific titles you are supposed to either:
* go with whatever NVIDIA sets in their profile, but NVIDIA is ... not interested in ReBAR, or
* otherwise try a few times for yourself with different BAR sizes.

I noticed NVIDIA goes with 1.5 GiB BAR for most titles. But there are titles that benefit if you set a larger BAR size, up to the maximum available.

So if there is one title that you are interested in, because it's what you play, it is definitely worth making a number of tries to find a good BAR size for you.

**Comment by LsdMax** [comment]
On cards with small amounts of memory, do not enable more than 1 gigabyte (nvidia inspector). There will be a memory overflow and it will get even worse and the video card will have to erase the full memory. Technically, the computer only remembers the last 4 requests of 256 megabytes.

**Comment by Fortio1** [comment]
Im curious, what does ReBAR options do? And is it best to enable as in - Enable - globally, or should I Enable globally and separately for lets say Hogwarts Legacy, 0x000000?
